### DISCUSSION TITLE
How to start Directus under AWS Elastic Beanstalk

### DISCUSSION THREAD

**Discussion by i-pasha** [directus/directus#5972] [Archive]
Heyo,
I have been struggling for days on my task to deploy directus on a beanstalk server using nodejs.
I have followed https://docs.directus.io/guides/installation/cli/ and started the server locally using npx directus start. No problems.
After I upload that exact code to codecommit and connect that with my beanstalk instance the server refuses to start. Here the error message:

```
100.0 % of the requests are failing with HTTP 5xx.
Process default has been unhealthy for 13 days (Target.ResponseCodeMismatch).
```

here is my configuration:

```
####################################################################################################
## General

PORT=8055
PUBLIC_URL="_URL_"
<<<<<<< HEAD

####################################################################################################
## Database

DB_CLIENT="mysql"
DB_HOST="_HOST_"
DB_PORT="3306"
=======
LOG_LEVEL="info"
LOG_STYLE="pretty"

####################################################################################################
# Database
DB_CLIENT="mysql"
DB_HOST="_URL_"
DB_PORT=3306
>>>>>>> 692774adf46e1e933951ga04592b6ba7a57ae00f
DB_DATABASE="directus"
DB_USER="_USER_"
DB_PASSWORD="_PASSWORD_"


####################################################################################################
## Rate Limiting

RATE_LIMITER_ENABLED=false
RATE_LIMITER_STORE=memory
RATE_LIMITER_POINTS=25
RATE_LIMITER_DURATION=1

####################################################################################################
## Cache

CACHE_ENABLED=false

####################################################################################################
## File Storage

STORAGE_LOCATIONS="local"
STORAGE_LOCAL_DRIVER="local"
STORAGE_LOCAL_ROOT="./uploads"

####################################################################################################
## Security

KEY="_KEY_"
SECRET="_SECRET_"


ACCESS_TOKEN_TTL="15m"
REFRESH_TOKEN_TTL="7d"
REFRESH_TOKEN_COOKIE_SECURE=false
REFRESH_TOKEN_COOKIE_SAME_SITE="lax"

####################################################################################################
## SSO (OAuth) Providers

OAUTH_PROVIDERS=""

####################################################################################################
## Extensions

EXTENSIONS_PATH="./extensions"

####################################################################################################
## Email

EMAIL_FROM="no-reply@directus.io"
EMAIL_TRANSPORT="sendmail"
EMAIL_SENDMAIL_NEW_LINE="unix"
EMAIL_SENDMAIL_PATH="/usr/sbin/sendmail"
```

The ">>>>>..." and "HEAD..." values were automatically generated but dont seem to affect the result in the local instance. 
The mysql instance has been set up correctly.
Am I missing something obvious? How many ressouces does directus need at least?

**Comment by rijkvanzanten** [comment]
> The ">>>>>..." and "HEAD..." values were automatically generated but dont seem to affect the result

This seems like a merge conflict that was merged in. You'd definitely want to clean that up first, as in this case you're already defining `DB_HOST` twice for example, which might cause trouble elsewhere.

The message you shared is AWSs health status, but doesn't tell us anything about what caused those 500s. Can you pull up the application logs for Directus itself? That should shed some light on why it's returning 500s for everything

**Reply by i-pasha** [reply]
Hi, I have cleared the junk out of the .env file.
I have created an ssh connetion to that server. The location where my code is deployed features a "Procfile" which contains "web: npm start" which is not present in my codecommit.
 Where are the application logs stored?

**Reply by rijkvanzanten** [reply]
I believe there's a dedicated tab in AWS that allows you to download the logs, see https://docs.aws.amazon.com/elasticbeanstalk/latest/dg/using-features.logging.html

**Reply by i-pasha** [reply]
```
----------------------------------------
/var/log/eb-engine.log
----------------------------------------
2021/06/01 14:57:02.885724 [INFO] Running command /bin/sh -c systemctl start aws-eb.target
2021/06/01 14:57:02.889556 [INFO] Running command /bin/sh -c systemctl enable healthd.service
2021/06/01 14:57:02.954116 [INFO] Running command /bin/sh -c systemctl show -p PartOf healthd.service
2021/06/01 14:57:02.958697 [INFO] Running command /bin/sh -c systemctl is-active healthd.service
2021/06/01 14:57:02.962006 [INFO] Running command /bin/sh -c systemctl start healthd.service
2021/06/01 14:57:02.992462 [INFO] Copying file /opt/elasticbeanstalk/config/private/healthd/healthd_logformat.conf to /var/proxy/staging/nginx/conf.d/healthd_logformat.conf
2021/06/01 14:57:02.994249 [INFO] Copying file /opt/elasticbeanstalk/config/private/healthd/healthd_nginx.conf to /var/proxy/staging/nginx/conf.d/elasticbeanstalk/healthd.conf
2021/06/01 14:57:02.995544 [INFO] Executing instruction: configure log streaming
2021/06/01 14:57:02.995556 [INFO] log streaming is not enabled
2021/06/01 14:57:02.995561 [INFO] disable log stream
2021/06/01 14:57:02.995582 [INFO] Running command /bin/sh -c systemctl show -p PartOf amazon-cloudwatch-agent.service
2021/06/01 14:57:03.002355 [INFO] Running command /bin/sh -c systemctl stop amazon-cloudwatch-agent.service
2021/06/01 14:57:03.006988 [INFO] Executing instruction: GetToggleForceRotate
2021/06/01 14:57:03.007005 [INFO] Checking if logs need forced rotation
2021/06/01 14:57:03.007027 [INFO] Running command /bin/sh -c /opt/aws/bin/cfn-get-metadata -s arn:ARN -r AWSEBAutoScalingGroup --region eu-central-1
2021/06/01 14:57:03.293410 [INFO] Running command /bin/sh -c /opt/aws/bin/cfn-get-metadata -s arn:ARN -r AWSEBBeanstalkMetadata --region eu-central-1
2021/06/01 14:57:03.577869 [INFO] Generating rsyslog config from Procfile
2021/06/01 14:57:03.579261 [INFO] Running command /bin/sh -c systemctl restart rsyslog.service
2021/06/01 14:57:03.602457 [INFO] Executing instruction: PostBuildEbExtension
2021/06/01 14:57:03.602479 [INFO] Starting executing the config set Infra-EmbeddedPostBuild.
2021/06/01 14:57:03.602504 [INFO] Running command /bin/sh -c /opt/aws/bin/cfn-init -s arn:ARN -r AWSEBAutoScalingGroup --region eu-central-1 --configsets Infra-EmbeddedPostBuild
2021/06/01 14:57:03.908160 [INFO] Finished executing the config set Infra-EmbeddedPostBuild.

2021/06/01 14:57:03.908188 [INFO] Executing instruction: CleanEbExtensions
2021/06/01 14:57:03.908209 [INFO] Cleaned ebextensions subdirectories from app staging directory.
2021/06/01 14:57:03.908214 [INFO] Executing instruction: RunAppDeployPreDeployHooks
2021/06/01 14:57:03.908230 [INFO] The dir .platform/hooks/predeploy/ does not exist in the application. Skipping this step...
2021/06/01 14:57:03.908235 [INFO] Executing instruction: stop X-Ray
2021/06/01 14:57:03.908264 [INFO] stop X-Ray ...
2021/06/01 14:57:03.908279 [INFO] Running command /bin/sh -c systemctl show -p PartOf xray.service
2021/06/01 14:57:03.913855 [WARN] stopProcess Warning: process xray is not registered 
2021/06/01 14:57:03.913878 [INFO] Running command /bin/sh -c systemctl stop xray.service
2021/06/01 14:57:03.918703 [INFO] Executing instruction: stop proxy
2021/06/01 14:57:03.918726 [INFO] Running command /bin/sh -c systemctl show -p PartOf httpd.service
2021/06/01 14:57:03.925256 [WARN] deregisterProcess Warning: process httpd is not registered, skipping...

2021/06/01 14:57:03.925277 [INFO] Running command /bin/sh -c systemctl show -p PartOf nginx.service
2021/06/01 14:57:03.931414 [WARN] deregisterProcess Warning: process nginx is not registered, skipping...

2021/06/01 14:57:03.931430 [INFO] Executing instruction: FlipApplication
2021/06/01 14:57:03.931434 [INFO] Fetching environment variables...
2021/06/01 14:57:03.931442 [INFO] setting default port 8080 to application
2021/06/01 14:57:03.931537 [INFO] Purge old process...
2021/06/01 14:57:03.931571 [INFO] Register application processes...
2021/06/01 14:57:03.931579 [INFO] Registering the proc: web

2021/06/01 14:57:03.931590 [INFO] Running command /bin/sh -c systemctl show -p PartOf web.service
2021/06/01 14:57:03.937359 [INFO] Running command /bin/sh -c systemctl daemon-reload
2021/06/01 14:57:03.984042 [INFO] Running command /bin/sh -c systemctl reset-failed
2021/06/01 14:57:03.987594 [INFO] Running command /bin/sh -c systemctl is-enabled eb-app.target
2021/06/01 14:57:03.990862 [INFO] Copying file /opt/elasticbeanstalk/config/private/aws-eb.target to /etc/systemd/system/eb-app.target
2021/06/01 14:57:03.992450 [INFO] Running command /bin/sh -c systemctl enable eb-app.target
2021/06/01 14:57:04.054029 [INFO] Running command /bin/sh -c systemctl start eb-app.target
2021/06/01 14:57:04.057799 [INFO] Running command /bin/sh -c systemctl enable web.service
2021/06/01 14:57:04.117957 [INFO] Running command /bin/sh -c systemctl show -p PartOf web.service
2021/06/01 14:57:04.122348 [INFO] Running command /bin/sh -c systemctl is-active web.service
2021/06/01 14:57:04.125821 [INFO] Running command /bin/sh -c systemctl start web.service
2021/06/01 14:57:04.151640 [INFO] Executing instruction: start X-Ray
2021/06/01 14:57:04.151659 [INFO] X-Ray is not enabled.
2021/06/01 14:57:04.151665 [INFO] Executing instruction: start proxy with new configuration
2021/06/01 14:57:04.151687 [INFO] Running command /bin/sh -c /usr/sbin/nginx -t -c /var/proxy/staging/nginx/nginx.conf
2021/06/01 14:57:04.180335 [INFO] Running command /bin/sh -c cp -rp /var/proxy/staging/nginx/* /etc/nginx
2021/06/01 14:57:04.183933 [INFO] Running command /bin/sh -c systemctl show -p PartOf nginx.service
2021/06/01 14:57:04.191114 [INFO] Running command /bin/sh -c systemctl daemon-reload
2021/06/01 14:57:04.250877 [INFO] Running command /bin/sh -c systemctl reset-failed
2021/06/01 14:57:04.254460 [INFO] Running command /bin/sh -c systemctl show -p PartOf nginx.service
2021/06/01 14:57:04.259384 [INFO] Running command /bin/sh -c systemctl is-active nginx.service
2021/06/01 14:57:04.262715 [INFO] Running command /bin/sh -c systemctl start nginx.service
2021/06/01 14:57:04.349139 [INFO] Executing instruction: configureSqsd
2021/06/01 14:57:04.349158 [INFO] This is a web server environment instance, skip configure sqsd daemon ...
2021/06/01 14:57:04.349164 [INFO] Executing instruction: startSqsd
2021/06/01 14:57:04.349168 [INFO] This is a web server environment instance, skip start sqsd daemon ...
2021/06/01 14:57:04.349173 [INFO] Executing instruction: Track pids in healthd
2021/06/01 14:57:04.349178 [INFO] This is an enhanced health env...
2021/06/01 14:57:04.349192 [INFO] Running command /bin/sh -c systemctl show -p ConsistsOf aws-eb.target | cut -d= -f2
2021/06/01 14:57:04.353675 [INFO] cfn-hup.service healthd.service nginx.service

2021/06/01 14:57:04.353697 [INFO] Running command /bin/sh -c systemctl show -p ConsistsOf eb-app.target | cut -d= -f2
2021/06/01 14:57:04.358008 [INFO] web.service

2021/06/01 14:57:04.358155 [INFO] Executing instruction: RunAppDeployPostDeployHooks
2021/06/01 14:57:04.358169 [INFO] The dir .platform/hooks/postdeploy/ does not exist in the application. Skipping this step...
2021/06/01 14:57:04.358174 [INFO] Executing cleanup logic
2021/06/01 14:57:04.360936 [INFO] Platform Engine finished execution on command: self-startup

----------------------------------------
/var/log/nginx/access.log
----------------------------------------
128.1.248.26 - - [01/Jun/2021:15:06:17 +0000] "GET /Telerik.Web.UI.WebResource.axd?type=rau HTTP/1.1" 502 559 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/60.0.3112.113 Safari/537.36" "-"
160.178.89.113 - - [01/Jun/2021:15:09:47 +0000] "GET /.env HTTP/1.1" 502 559 "-" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.129 Safari/537.36" "-"
160.178.89.113 - - [01/Jun/2021:15:09:47 +0000] "POST / HTTP/1.1" 502 559 "-" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.129 Safari/537.36" "-"
37.201.144.193 - - [01/Jun/2021:15:12:14 +0000] "GET / HTTP/1.1" 502 559 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/90.0.4430.212 Safari/537.36" "-"
37.201.144.193 - - [01/Jun/2021:15:12:15 +0000] "GET /favicon.ico HTTP/1.1" 502 559 "http://HOSTURL/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/90.0.4430.212 Safari/537.36" "-"
167.248.133.40 - - [01/Jun/2021:15:27:38 +0000] "GET / HTTP/1.1" 502 157 "-" "-" "-"
167.248.133.40 - - [01/Jun/2021:15:27:39 +0000] "GET / HTTP/1.1" 502 157 "-" "Mozilla/5.0 (compatible; CensysInspect/1.1; +https://about.censys.io/)" "-"
37.201.144.193 - - [01/Jun/2021:16:02:33 +0000] "GET / HTTP/1.1" 502 559 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/90.0.4430.212 Safari/537.36" "-"
37.201.144.193 - - [01/Jun/2021:16:02:33 +0000] "GET /favicon.ico HTTP/1.1" 502 559 "http://HOSTURL/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/90.0.4430.212 Safari/537.36" "-"


----------------------------------------
/var/log/nginx/error.log
----------------------------------------
2021/06/01 15:06:17 [error] 3896#0: *1 connect() failed (111: Connection refused) while connecting to upstream, client: 128.1.248.26, server: , request: "GET /Telerik.Web.UI.WebResource.axd?type=rau HTTP/1.1", upstream: "http://127.0.0.1:8080/Telerik.Web.UI.WebResource.axd?type=rau", host: "HOSTIP"
2021/06/01 15:09:47 [error] 3896#0: *4 connect() failed (111: Connection refused) while connecting to upstream, client: 160.178.89.113, server: , request: "GET /.env HTTP/1.1", upstream: "http://127.0.0.1:8080/.env", host: "HOSTIP"
2021/06/01 15:09:47 [error] 3896#0: *6 connect() failed (111: Connection refused) while connecting to upstream, client: 160.178.89.113, server: , request: "POST / HTTP/1.1", upstream: "http://127.0.0.1:8080/", host: "HOSTIP"
2021/06/01 15:12:14 [error] 3896#0: *8 connect() failed (111: Connection refused) while connecting to upstream, client: 37.201.144.193, server: , request: "GET / HTTP/1.1", upstream: "http://127.0.0.1:8080/", host: "HOSTURL"
2021/06/01 15:12:15 [error] 3896#0: *8 connect() failed (111: Connection refused) while connecting to upstream, client: 37.201.144.193, server: , request: "GET /favicon.ico HTTP/1.1", upstream: "http://127.0.0.1:8080/favicon.ico", host: "HOSTURL", referrer: "http://HOSTURL/"
2021/06/01 15:27:38 [error] 3896#0: *13 connect() failed (111: Connection refused) while connecting to upstream, client: 167.248.133.40, server: , request: "GET / HTTP/1.1", upstream: "http://127.0.0.1:8080/", host: "HOSTIP:80"
2021/06/01 15:27:39 [error] 3896#0: *15 connect() failed (111: Connection refused) while connecting to upstream, client: 167.248.133.40, server: , request: "GET / HTTP/1.1", upstream: "http://127.0.0.1:8080/", host: "HOSTIP"
2021/06/01 16:02:33 [error] 3896#0: *17 connect() failed (111: Connection refused) while connecting to upstream, client: 37.201.144.193, server: , request: "GET / HTTP/1.1", upstream: "http://127.0.0.1:8080/", host: "HOSTURL"
2021/06/01 16:02:33 [error] 3896#0: *17 connect() failed (111: Connection refused) while connecting to upstream, client: 37.201.144.193, server: , request: "GET /favicon.ico HTTP/1.1", upstream: "http://127.0.0.1:8080/favicon.ico", host: "HOSTURL", referrer: "http://HOSTURL/"

```

ARN, HOSTIP and HOSTURL are masked.

**Reply by rijkvanzanten** [reply]
Looks like directus itself isn't running at all. How are you starting the node process? You mentioned something about a Procfile with a non-existing command?

**Reply by i-pasha** [reply]
I am not sure. my package.json says: 

> "scripts": {
    "start": "npx directus start"
  },

That should do the trick. Or should I define the Procfile myself?

**Reply by i-pasha** [reply]
I think I have pinpointed the problem:

> Error: connect ETIMEDOUT
    at Connection._handleConnectTimeout (/home/ec2-user/.npm/_npx/4996/lib/node_modules/directus/node_modules/mysql/lib/Connection.js:409:13)
    at Object.onceWrapper (events.js:482:28)
    at Socket.emit (events.js:376:20)
    at Socket._onTimeout (net.js:483:8)
    at listOnTimeout (internal/timers.js:555:17)
    at processTimers (internal/timers.js:498:7)
    --------------------
    at Protocol._enqueue (/home/ec2-user/.npm/_npx/4996/lib/node_modules/directus/node_modules/mysql/lib/protocol/Protocol.js:144:48)
    at Protocol.handshake (/home/ec2-user/.npm/_npx/4996/lib/node_modules/directus/node_modules/mysql/lib/protocol/Protocol.js:51:23)
    at Connection.connect (/home/ec2-user/.npm/_npx/4996/lib/node_modules/directus/node_modules/mysql/lib/Connection.js:116:18)
    at /home/ec2-user/.npm/_npx/4996/lib/node_modules/directus/node_modules/knex/lib/dialects/mysql/index.js:56:18
    at new Promise (<anonymous>)
    at Client_MySQL.acquireRawConnection (/home/ec2-user/.npm/_npx/4996/lib/node_modules/directus/node_modules/knex/lib/dialects/mysql/index.js:51:12)
    at create (/home/ec2-user/.npm/_npx/4996/lib/node_modules/directus/node_modules/knex/lib/client.js:237:39)
    at processTicksAndRejections (internal/process/task_queues.js:95:5) {
  errorno: 'ETIMEDOUT',
  code: 'ETIMEDOUT',
  syscall: 'connect',
  fatal: true
}


After i trigger npx directus start over the cli this error is displayed. At the same time the beanstalk alert says that "100% of CPU is in use" while the startup is in progress. (I am using the t2.micro at the moment). I can connect to the database mysqlf via mySQL Workbench no problem. Are there other aws related configuration I need to take account for to ensure this connection?

**Reply by rijkvanzanten** [reply]
Does your RDS have any specific access requirements? AWSs permissions are very powerful, but also pretty confusing to setup. Make sure your Elastic Beanstalk instance has access to RDS, and that you're using the correct database credentials in the configured environment variables 👍🏻

**Reply by i-pasha** [reply]
I have set up a new database connection through the beanstalk gui.
After I adjusted the new db url and punched in "npx directus start" this logs appears:

```
> internal/bootstrap/switches/does_own_process_state.js:130
    cachedCwd = rawMethods.cwd();
                           ^

Error: ENOENT: no such file or directory, uv_cwd
    at process.wrappedCwd [as cwd] (internal/bootstrap/switches/does_own_process_state.js:130:28)
    at npx (/opt/elasticbeanstalk/node-install/node-v14.17.0-linux-x64/lib/node_modules/npm/node_modules/libnpx/index.js:39:31)
    at Object.<anonymous> (/opt/elasticbeanstalk/node-install/node-v14.17.0-linux-x64/lib/node_modules/npm/bin/npx-cli.js:8:1)
    at Module._compile (internal/modules/cjs/loader.js:1068:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1097:10)
    at Module.load (internal/modules/cjs/loader.js:933:32)
    at Function.Module._load (internal/modules/cjs/loader.js:774:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:72:12)
    at internal/main/run_main_module.js:17:47 {
  errno: -2,
  code: 'ENOENT',
  syscall: 'uv_cwd'
}
```

**Reply by rijkvanzanten** [reply]
That error seems to have been thrown by `npx`, not directus. From a quick google search, it seems like the most common culprit is trying to run `npx <whatever>` in a non-existing folder?

**Reply by i-pasha** [reply]
Sorry that was caused because of an old ssh connection.
Re-entering it surely helped. After I punched the script this is displayed:

> Error: ER_BAD_DB_ERROR: Unknown database 'directus'
    at Handshake.Sequence._packetToError (/home/ec2-user/.npm/_npx/30318/lib/node_modules/directus/node_modules/mysql/lib/protocol/sequences/Sequence.js:47:14)
    at Handshake.ErrorPacket (/home/ec2-user/.npm/_npx/30318/lib/node_modules/directus/node_modules/mysql/lib/protocol/sequences/Handshake.js:123:18)
    at Protocol._parsePacket (/home/ec2-user/.npm/_npx/30318/lib/node_modules/directus/node_modules/mysql/lib/protocol/Protocol.js:291:23)
    at Parser._parsePacket (/home/ec2-user/.npm/_npx/30318/lib/node_modules/directus/node_modules/mysql/lib/protocol/Parser.js:433:10)
    at Parser.write (/home/ec2-user/.npm/_npx/30318/lib/node_modules/directus/node_modules/mysql/lib/protocol/Parser.js:43:10)
    at Protocol.write (/home/ec2-user/.npm/_npx/30318/lib/node_modules/directus/node_modules/mysql/lib/protocol/Protocol.js:38:16)
    at Socket.<anonymous> (/home/ec2-user/.npm/_npx/30318/lib/node_modules/directus/node_modules/mysql/lib/Connection.js:88:28)
    at Socket.<anonymous> (/home/ec2-user/.npm/_npx/30318/lib/node_modules/directus/node_modules/mysql/lib/Connection.js:526:10)
    at Socket.emit (events.js:376:20)
    at addChunk (internal/streams/readable.js:309:12)
    --------------------
    at Protocol._enqueue (/home/ec2-user/.npm/_npx/30318/lib/node_modules/directus/node_modules/mysql/lib/protocol/Protocol.js:144:48)
    at Protocol.handshake (/home/ec2-user/.npm/_npx/30318/lib/node_modules/directus/node_modules/mysql/lib/protocol/Protocol.js:51:23)
    at Connection.connect (/home/ec2-user/.npm/_npx/30318/lib/node_modules/directus/node_modules/mysql/lib/Connection.js:116:18)
    at /home/ec2-user/.npm/_npx/30318/lib/node_modules/directus/node_modules/knex/lib/dialects/mysql/index.js:56:18
    at new Promise (<anonymous>)
    at Client_MySQL.acquireRawConnection (/home/ec2-user/.npm/_npx/30318/lib/node_modules/directus/node_modules/knex/lib/dialects/mysql/index.js:51:12)
    at create (/home/ec2-user/.npm/_npx/30318/lib/node_modules/directus/node_modules/knex/lib/client.js:237:39)
    at processTicksAndRejections (internal/process/task_queues.js:95:5) {
  code: 'ER_BAD_DB_ERROR',
  errno: 1049,
  sqlMessage: "Unknown database 'directus'",
  sqlState: '42000',
  fatal: true
}

I inspected the database through mysql workbench and it looked fine(?)
![db](https://user-images.githubusercontent.com/83359988/120697054-88e49480-c4ad-11eb-8f48-849edfb348d8.jpg)
We are definitely making progress and I am really really thankful for your help so far :)

**Reply by rijkvanzanten** [reply]
Gotchu! Getting there :) 

Are you connecting as the same user in the same database in Workbench? Maybe the user you're using in Elastic Beanstalk doesn't have read access to the `directus` database?

**Reply by i-pasha** [reply]
It is the same user and is publicly accessible. The only thing different is the url which I have changed in the .env file.
As far as I understand the application should have been able to access that database to setup the tables. It should be the same connection right?

**Comment by i-pasha** [comment]
@rijkvanzanten  Okay, I have made quite the progress: I have: 
- deleted the directus database in my mysql db.
- re-initiated the setup locally using npx directus init. That fixed the "Unknown database 'directus'"
- re-started the beanstalk instance
- exec-ed into the ec2 instance using ssh
- manually triggered npx directus start (After a few failed attempts I additionally used that line with sudo and as root; both without success)
- After a 3rd time I used that line as the ec2-user without root back-to-back I finally got "Server started at port 8055". Yay ^^
- Loading my Webpage only displays a white page (the elements don't seem to load).
- (Now I don't have to manually type in npx directus start. It seems to do that automatically when the beanstalk instance starts. Finally)

My current load balancer is the following:
![2](https://user-images.githubusercontent.com/83359988/120875116-6f754280-c5aa-11eb-8207-c2db3d1f29bc.jpeg)
The first line describes the environment url elasticbeanstalk automatically generates and the second the one I use with ssl (already added in cert manager and the https://... URL is saved under PUBLIC_URL in .env).
instead of "instance-port = 80" should it be 8055?
We are nearly there :D

**Reply by rijkvanzanten** [reply]
> instead of "instance-port = 80" should it be 8055?

I think so, yeah!

**Reply by i-pasha** [reply]
That was it. Thank you for your endurance with my issue ^^
